src/legacy/core_plugins/kibana/migrations.js

```diff
@@ -57,6 +57,34 @@ function migrateIndexPattern(doc) {
   doc.attributes.kibanaSavedObjectMeta.searchSourceJSON = JSON.stringify(searchSource);
 }
 
+function removeDateHistogramTimeZones(doc) {
+  const newDoc = cloneDeep(doc);
+  const visStateJSON = get(doc, 'attributes.visState');
+  if (visStateJSON) {
+    let visState;
+    try {
+      visState = JSON.parse(visStateJSON);
+    } catch (e) {
+      // Let it go, the data is invalid and we'll leave it as is
+    }
+    if (visState && visState.aggs) {
+      visState.aggs.forEach(agg => {
+        // We're checking always for the existance of agg.params here. This should always exist, but better
+        // be safe then sorry during migrations.
+        if (agg.type === 'date_histogram' && agg.params) {
+          delete agg.params.time_zone;
+        }
+
+        if (get(agg, 'params.customBucket.type', null) === 'date_histogram' && agg.params.customBucket.params) {
+          delete agg.params.customBucket.params.time_zone;
+        }
```

Comment:
nit: lodash `omit` could save you from some conditionals here, e.g.
```javascript
if (agg.type === 'date_histogram' || get(agg, 'params.customBucket.type') === 'date_histogram') {
  agg.params = omit(agg.params, ['time_zone', 'customBucket.params.time_zone']);
}
```

Reply:
Would you prefer that? I find it actually harder to read, then the more explicit version :-)

Reply:
I think either is fine. Go with what you think is most obvious.

Reply:
I marked it as a nit for a reason 😉  Totally up to your preference here, I only suggested it because `omit` will automatically skip things that are undefined.

Reply:
(And FWIW I had no issues groking the code as it was currently written)